src/components/Dropdown.js

```diff
@@ -0,0 +1,260 @@
+
+export const MenuDivider = glamorous.div({
+  display: 'block',
+  width: '100%',
+  height: 0,
+  borderTop: `1px solid ${toAlpha(colors.gray[9], colors.black)}`,
+  margin: joinSpacing(spacing[0], 0),
+})
+
+class Dropdown extends Component {
+  static propTypes = {
+    renderMenuButton: func.isRequired,
+    align: oneOf(['right', 'left']),
+    minWidth: oneOfType([number, string]),
+  }
+
+  static defaultProps = {
+    align: 'right',
+    minWidth: 'auto',
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.menuButtonRef = React.createRef()
+    this.menuRef = React.createRef()
+
+    this.state = {
+      isOpen: false,
+    }
+  }
+
+  componentDidMount() {
+    this.menuItems = this.menuRef.current.querySelectorAll('[role^="menuitem"]')
+
+    if (this.menuItems.length < 1) {
+      throw new Error('No menu items')
+    }
+
+    const activeMenuItems = Array.prototype.filter.call(
+      this.menuItems,
+      item => !item.disabled,
+    )
+
+    // Disable menu button if all menu items are disabled
+    if (activeMenuItems.length < 1) {
+      this.menuButtonRef.current.disabled = true
+      return
+    }
+
+    this.firstItem = this.menuItems[0]
+    this.lastItem = this.menuItems[this.menuItems.length - 1]
+
+    Array.prototype.forEach.call(this.menuItems, menuItem => {
+      menuItem.addEventListener('keydown', this.handleMenuItemKeyDown)
+    })
+
+    this.menuButtonRef.current.addEventListener(
+      'keydown',
+      this.handleMenuButtonKeyDown,
+    )
+  }
+
+  componentWillUnmount() {
+    Array.prototype.forEach.call(this.menuItems, menuItem => {
+      menuItem.removeEventListener('keydown', this.handleMenuItemKeyDown)
+    })
+
+    this.menuButtonRef.current.removeEventListener(
+      'keydown',
+      this.handlMenuButtonKeyDown,
+    )
+
+    document.removeEventListener('click', this.close)
+  }
+
+  handleMenuButtonKeyDown = event => {
+    if (event.key === 'ArrowDown') {
+      event.preventDefault()
+      if (this.state.isOpen) {
+        this.menuRef.current
+          .querySelector('[role^="menuitem"]:not([disabled])')
+          .focus()
+      } else {
+        this.open()
+      }
+    }
+
+    if (event.key === 'Escape' || event.key === 'Tab') {
+      this.close()
+    }
+
+    if (event.key === 'Escape') {
+      event.preventDefault()
+      this.menuButtonRef.current.focus()
+    }
+  }
+
+  handleMenuItemKeyDown = event => {
+    if (event.key === 'ArrowDown') {
+      event.preventDefault()
+      this.focusNext(event.target, this.firstItem)
+    }
+
+    if (event.key === 'ArrowUp') {
+      event.preventDefault()
+      this.focusNext(event.target, this.lastItem)
+    }
+
+    if (event.key === 'Escape' || event.key === 'Tab') {
+      this.close()
+    }
+
+    if (event.key === 'Escape') {
+      event.preventDefault()
+      this.menuButtonRef.current.focus()
+    }
+  }
+
+  focusNext = (currentItem, startItem) => {
+    // Determine which item is the startItem (first or last)
```

Comment:
This is nitpicky, but I feel like we could write this comment a little better.
Maybe something more like`// Determine if we need to move up to down depending on whether the startItem is the first or last item`.
Not saying that's great, just a little more explicit.

Reply:
`// startItem will either be the first or the last item. If it is the first item, then we are going down`